### PR TITLE
Fix health DTOs Jackson deserialization

### PR DESCRIPTION
**Pull request by nathanschile**
What was changed? Why is this necessary?
----------------------------------------

* Fixes health DTOs Jackson deserialization by adding @JsonDeserialize and @JsonCreator

How was it tested?
----

* Unit test

How to test
----

*This is bare minimum acceptable testing*

- [ ] `mvn clean install -U`

### CODE REVIEW

File: health/api/src/main/java/com/cerner/beadledom/health/dto/BuildDto.java [hunk 1]
```diff
@@ -15,8 +17,9 @@
  *
  * @since 1.4
  */
-@ApiModel(description = "Indicates the list of dependencies of this service.")
 @AutoValue
+@ApiModel(description = "Indicates the list of dependencies of this service.")
+@JsonDeserialize(builder = AutoValue_BuildDto.Builder.class)
```
Comment (johnlcox) [inline health/api/src/main/java/com/cerner/beadledom/health/dto/BuildDto.java]: `BuildDto.Builder.class`

File: health/api/src/main/java/com/cerner/beadledom/health/dto/BuildDto.java [hunk 2]
```diff
@@ -15,8 +17,9 @@
  *
  * @since 1.4
  */
-@ApiModel(description = "Indicates the list of dependencies of this service.")
 @AutoValue
+@ApiModel(description = "Indicates the list of dependencies of this service.")
```
Comment (johnlcox) [inline health/api/src/main/java/com/cerner/beadledom/health/dto/BuildDto.java]: Make sure the order of these annotations is consistent for all the health DTOs. It seems like most had `JsonDeserialize` before `ApiModel`.

File: health/api/src/test/scala/com/cerner/beadledom/health/dto/HealthDependenciesDtoSpec.scala [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+                         .setName("name")
+                         .setMessage("message")
+                         .setPrimary(true)
+                         .setLinks(LinksDto.builder().setSelf("link").build())
+                         .build()
+
+  var healthDependenciesDto = HealthDependenciesDto.builder()
+                              .setDependencies(List(healthDependency).asJava).build()
+
+  it must behave like serializableByJackson(healthDependenciesDto)
```
Comment (johnlcox) [inline health/api/src/test/scala/com/cerner/beadledom/health/dto/HealthDependenciesDtoSpec.scala]: Seeing this now in an actual `behave` statement, I wonder if the behavior method would be better named as `aJacksonSerializableObject`

File: health/api/src/main/java/com/cerner/beadledom/health/dto/BuildDto.java (hunk 1 above)
Comment (nathanschile) [reply health/api/src/main/java/com/cerner/beadledom/health/dto/BuildDto.java]: Updated, https://github.com/cerner/beadledom/pull/100/commits/5b462705934358b648b7e9f6b532f7364b1d0aa6

File: health/api/src/main/java/com/cerner/beadledom/health/dto/BuildDto.java (hunk 2 above)
Comment (nathanschile) [reply health/api/src/main/java/com/cerner/beadledom/health/dto/BuildDto.java]: Fixed ordering, https://github.com/cerner/beadledom/pull/100/commits/16bb80cf2621d60970f0ad16e06a7708768dd997

File: health/api/src/test/scala/com/cerner/beadledom/health/dto/HealthDependenciesDtoSpec.scala (hunk 3 above)
Comment (nathanschile) [reply health/api/src/test/scala/com/cerner/beadledom/health/dto/HealthDependenciesDtoSpec.scala]: I like it, https://github.com/cerner/beadledom/pull/100/commits/ac509b10974ebd5fe955358ff52a3000348ee107

File: health/api/src/main/java/com/cerner/beadledom/health/dto/BuildDto.java [hunk 4]
```diff
@@ -41,22 +44,31 @@ public static Builder builder() {
...
   public abstract static class Builder {
 
+    @JsonCreator
+    private static BuildDto.Builder create() {
+      return BuildDto.builder();
+    }
+
     abstract Builder setArtifactName(Optional<String> artifactName);
 
+    @JsonProperty("artifactName")
```
Comment (sparuvu) [inline health/api/src/main/java/com/cerner/beadledom/health/dto/BuildDto.java]: why do we need to give the property name explicitly. I thought Jackson automatically infers from the method name.

File: health/api/src/test/scala/com/cerner/beadledom/health/dto/HealthDependencyDtoSpec.scala [hunk 5]
```diff
@@ -53,4 +54,16 @@ class HealthDependencyDtoSpec extends FunSpec with MustMatchers {
...
+    .setHealthy(true)
+    .setId("1")
+    .setName("name")
+    .setMessage("message")
+    .setPrimary(true)
+    .setLinks(LinksDto.builder().setSelf("link").build())
+    .build()
+
+  it must behave like aJacksonSerializableObject(healthDependencyDto)
+
```
Comment (sparuvu) [inline health/api/src/test/scala/com/cerner/beadledom/health/dto/HealthDependencyDtoSpec.scala]: nit: empty line here and at other places.

File: health/api/src/main/java/com/cerner/beadledom/health/dto/BuildDto.java (hunk 4 above)
Comment (johnlcox) [reply health/api/src/main/java/com/cerner/beadledom/health/dto/BuildDto.java]: The Beadledom default object mapper uses underscores, but the health check needs to use camel case.

File: health/api/src/test/scala/com/cerner/beadledom/health/dto/HealthDependencyDtoSpec.scala (hunk 5 above)
Comment (nathanschile) [reply health/api/src/test/scala/com/cerner/beadledom/health/dto/HealthDependencyDtoSpec.scala]: Removed empty lines, https://github.com/cerner/beadledom/pull/100/commits/e90e469d9c9a4c65fb23ae4bbd01e37a90a39527

**Pull request merged**